js/ui/appDisplay.js

```diff
@@ -2478,27 +2478,6 @@ const AppIcon = new Lang.Class({
                     this.activateWindow(null);
                 }));
             }
-
-            if (Shell.AppSystem.get_default().lookup_app('org.gnome.Software.desktop')) {
```

Comment:
I believe the problem is that you land in GNOME Software in an invalid page instead of in the one for the application, as it happens with other apps (e.g. chromium, nautilus, duolingo...). @rshuler will know more about this, though, but in any case I'd leave it out for now as we didn't have this before anyway.